readthedocs/restapi/views/integrations.py

```diff
@@ -108,10 +128,19 @@ def get_integration(self):
         # in `WebhookView`
         if self.integration is not None:
             return self.integration
-        integration, _ = Integration.objects.get_or_create(
-            project=self.project,
-            integration_type=self.integration_type,
-        )
+        try:
+            integration = Integration.objects.get(
```

Comment:
Now that we are deprecating old webhooks, is this `get_integration` method that guesses still needed?

Reply:
I'm not sure, this is about webhooks that were created without a model. I guess that the old webhooks (using deprecated services) they could have a model?

Reply:
We should test this more. The old webhooks don't touch integrations at all, but I'm not sure why we need logic that tries to look up an integration without a `pk`. All integrations have a `pk` attached, and there should be no reason that we should need to look up an  integration without using a `pk`. At least,  I think...